### PR TITLE
getDestFilePath: handle non-existent (deleted/moved) files

### PR DESCRIPTION
**Pull request by jamescdavis**
https://github.com/broccolijs/broccoli-persistent-filter/blob/87738eb2bfedac6dabf31d3cf6b1eeb60287c361/src/index.ts#L422-L424

see failing test and fix commits

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -416,7 +416,7 @@ abstract class Filter extends Plugin {
       return false;
     }
 
-    return (entry || this.input.lstatSync(relativePath)).isDirectory();
+    return entry ? entry.isDirectory() : this.input.existsSync(relativePath) && this.input.lstatSync(relativePath).isDirectory();
```
Comment (rwjblue) [inline src/index.ts:419]: This is going to do two stat operations (one for input.existsSync and one for input.lstatSync), but I think we can actually avoid that. Additionally, this is mixing the concepts of stat and lstat (`existsSync` does a stat and therefore follows symlinks, but then we are actually using `lstat().isDirectory()`).

```js
if (entry !== undefined) {
  return entry.isDirectory();
} else {
  try {
    const stat = this.input.lstatSync(relativePath);
    return stat.isDirectory();
  } catch (error) {
    // if we get any other error, we really don't know what is going on so we need to rethrow
    if (error.code === 'ENOENT') {
      return false;
    }
    throw error;
  }
}
```

(Apologies for typos here, typed it out on mobile)

**Review by rwjblue — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 1 above)
Comment (jamescdavis) [reply src/index.ts:419]: lol, see: https://github.com/dfreeman/broccoli-graphql-filter/pull/20/files

**Review by rwjblue — APPROVED** [review]
Thanks @jamescdavis!

**Comment by rwjblue** [comment]
https://github.com/broccolijs/broccoli-persistent-filter/releases/tag/v3.1.2

**Pull request merged**